### PR TITLE
std: make c.versionCheck() a comptime-known function

### PR DESCRIPTION
**Pull request by nektro**
resolved a TODO 🎊

### CODE REVIEW

File: lib/std/c.zig [hunk 1]
```diff
@@ -5,29 +5,25 @@ const page_size = std.mem.page_size;
 const iovec = std.os.iovec;
 const iovec_const = std.os.iovec_const;
 
-/// The return type is `type` to force comptime function call execution.
-/// TODO: https://github.com/ziglang/zig/issues/425
 /// If not linking libc, returns struct{pub const ok = false;}
```
Comment (rootbeer) [inline lib/std/c.zig:8]: The comments about the return type being a struct are stale now.

File: lib/std/c.zig [hunk 2]
```diff
@@ -5,29 +5,25 @@ const page_size = std.mem.page_size;
...
-                    .gt, .eq => true,
-                    .lt => false,
-                };
-            } else {
-                break :blk false;
-            }
-        };
+pub inline fn versionCheck(comptime glibc_version: std.SemanticVersion) bool {
+    return comptime blk: {
+        if (!builtin.link_libc) break :blk false;
```
Comment (rootbeer) [inline lib/std/c.zig:15]: Might be more straight-forward to remove the `blk:` tags/break and just `return false` or `return true` in each branch?

File: lib/std/c.zig [hunk 3]
```diff
@@ -5,29 +5,25 @@ const page_size = std.mem.page_size;
...
+        if (builtin.abi.isMusl()) break :blk true;
+        if (builtin.target.isGnuLibC()) {
+            const ver = builtin.os.version_range.linux.glibc;
+            const order = ver.order(glibc_version);
+            break :blk switch (order) {
+                .gt, .eq => true,
+                .lt => false,
+            };
```
Comment (rootbeer) [inline lib/std/c.zig:23]: Simplify to just `return ver.order(glibc_version) != .lt`?

(It is a bit dense like this, but is how the version comparison checks are generally done elsewhere.)

**Review by rootbeer — APPROVED** [review]
LGTM!  Feel free to ignore my comments on the code itself if you like the current version.  (The stale comment really should be fixed I think.)

File: lib/std/c.zig (hunk 2 above)
Comment (nektro) [reply lib/std/c.zig:15]: it makes a function call which would make that branch runtime known, the comptime block is used to force the entire expression to be comptime

**Pull request merged**